cmd/keeper/keeper.go

```diff
@@ -1073,8 +1073,16 @@ func (p *PostgresKeeper) postgresKeeperSM(pctx context.Context) {
 				log.Errorw("failed to start instance", zap.Error(err))
 				return
 			}
-			// wait for the db having replyed all the wals
-			if err = pgm.WaitReady(cd.Cluster.DefSpec().SyncTimeout.Duration); err != nil {
+
+			deadline := time.Now().Add(cd.Cluster.DefSpec().SyncTimeout.Duration)
```

Comment:
deadline should be a time.Duration. You don't have to make it a time.Time adding it to time.Now() and then subtracting Time.Now in the below calls.